File: api/express/src/entity/Note.ts
import { Exclude, Expose } from 'class-transformer';

export default class Note {
  public readonly id: string;

  public readonly user_id: string;

  public readonly title: string;

  public readonly text: string;

  public readonly color: string;

  @Exclude()
  public readonly updatedAt: Date;

  @Expose({ name: 'updated_at' })
  getUpdatedAt(): Date {
    return this.updatedAt;
  }

  constructor(
    id: string,
    user_id: string,
    title: string,
    text: string,
    color: string,
    updatedAt: Date,
  ) {
    this.id = id;
    this.user_id = user_id;
    this.title = title;
    this.text = text;
    this.color = color;
    this.updatedAt = updatedAt;
  }
}

File: app/lib/domain/models/note.dart
final class Note {
  final String id;
  final String userId;
  final String title;
  final String text;
  final String updatedAt;
  final String color;

  const Note({
    required this.id,
    required this.userId,
    required this.title,
    required this.text,
    required this.updatedAt,
    required this.color,
  });
}
